salt/modules/mac_shadow.py

```diff
@@ -189,8 +160,49 @@ def get_last_change(name):
 
         salt '*' shadow.get_last_change admin
     '''
-    unix_timestamp = _get_account_policy_data_value(name, 'passwordLastSetTime')
-    return _convert_to_datetime(unix_timestamp)
+    try:
+        pwd.getpwnam(name)
+    except KeyError:
+        log.debug('User not found: {0}'.format(name))
+        raise CommandExecutionError('User not found: {0}'.format(name))
+
+    cmd = 'dscl . -readpl /Users/{0} ' \
+          'accountPolicyData {1}'.format(name, 'passwordLastSetTime')
+    ret = salt.utils.mac_utils.execute_return_result(cmd)
+
+    unix_timestamp = salt.utils.mac_utils.parse_return(ret)
+
+    date_text = _convert_to_datetime(unix_timestamp)
+
+    return date_text
+
+
+def get_login_failed_count(name):
+    '''
+    Get the the number of failed login attempts
+
+    :param str name: the username of the account
+
+    :return: The number of failed login attempts
+    :rtype: int
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' shadow.get_login_failed_count admin
+    '''
+    try:
+        pwd.getpwnam(name)
+    except KeyError:
+        log.debug('User not found: {0}'.format(name))
+        raise CommandExecutionError('User not found: {0}'.format(name))
```

Comment:
If we're going to do this, we need to document it so that a user knows to expect it as an error condition.